File: scala/greet.scala
class Greet(val who: String) {
    val name: String = who.toUpperCase().charAt(0) +
                       who.substring(1)

    def salute() {
        println("Hello " + name + "!")
    }
}

val g = new Greet("world") // create object
g.salute() // output "Hello World!"

File: java/Greet.java
public class Greet {
    private String name;

    public Greet(String who) {
        name = who.toUpperCase().charAt(0) +
            who.substring(1);
    }

    public void salute() {
        System.out.println("Hello " + name + "!");
    }

    public static void main(String... args) {
        final Greet g = new Greet("world");
        g.salute();
    }
}
